tokio/src/time/driver/mod.rs

```diff
@@ -248,6 +266,18 @@ where
 
         Ok(())
     }
+
+    cfg_test_util! {
+        fn did_wake(&self) -> bool {
+            self.did_wake.swap(false, Ordering::SeqCst)
+        }
+    }
+
+    cfg_not_test_util! {
+        fn did_wake(&self) -> bool {
+            unreachable!()
+        }
+    }
```

Comment:
It would be neat if we didn't need this.

Reply:
It's this way to avoid cfgs in the main code, which, IMO, is more confusing.

Reply:
Do you have thoughts on how it can be improved?

Reply:
If you factor out this snippet to a function, then you can define _that_ function twice with and without `test-util`. Then the version without `test-util` can just call `park_timeout` directly.
```Rust
if clock.is_paused() {
    self.park.park_timeout(Duration::from_secs(0))?;

    // Simulate advancing time
    clock.advance(duration);
    // If the time driver was woken, then the park completed
    // before the "duration" elapsed (usually caused by a
    // yield in `Runtime::block_on`). In this case, we don't
    // advance the clock.
    if !self.did_wake() {
        // Simulate advancing time
        clock.advance(duration);
    }
} else {
    self.park.park_timeout(duration)?;
}
```